### PR TITLE
opentracing/deprecation: Minor changelog cleanup

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @RyanTheOptimist

File: api/envoy/config/trace/v3/dynamic_ot.proto [hunk 1]
```diff
@@ -33,11 +33,11 @@ message DynamicOtConfig {
   string library = 1 [
     deprecated = true,
     (validate.rules).string = {min_len: 1},
-    (envoy.annotations.deprecated_at_minor_version) = "3.0"
+    (envoy.annotations.deprecated_at_minor_version) = "2.10"
```
Comment (RyanTheOptimist) [inline api/envoy/config/trace/v3/dynamic_ot.proto:36]: Is this an Envoy API version or an OT version? My understanding of this annotation is that it refers to an Envoy API version.

Comment (phlax) [reply api/envoy/config/trace/v3/dynamic_ot.proto:36]: i was just reading the relevant proto ...

afaict you are correct this refers to the proto version, altho given that there is no intention of increasing the api version im not sure that is what it _should_ be (separately its intended to refer to the version at which it becomes deprecated rather than the version at which it will be removed)

ill revert the version changes but i think the changelog change is correct

File: changelogs/current.yaml [hunk 2]
```diff
@@ -83,3 +80,6 @@ new_features:
     added support for lmove command.
 
 deprecated:
+- area: tracing
+  change: |
+    OpenTracing is deprecated and will be removed at version 2.10, since the upstream project has been abandoned.
```
Comment (RyanTheOptimist) [inline changelogs/current.yaml:85]: How 'bout we slightly rephrase this to be:
```
OpenTracing is deprecated and will be removed, since the upstream project has been abandoned. OpenTracing 2.10 is the last version which will be imported into Envoy.
```

Comment (phlax) [reply changelogs/current.yaml:85]: i think there is not much point putting a deprecation notice if it doesnt say _when_ the thing will be removed - its pretty unhelpful

i have put version 2.10 on the basis that this is the first version after the 6 month cutoff

Comment (phlax) [reply changelogs/current.yaml:85]: envoy version

Comment (phlax) [reply changelogs/current.yaml:85]: as discussed offline - im getting totally confused here - the removal version should be `1.30` not `2.10`

ill update

**Review by RyanTheOptimist — APPROVED** [review]
Thanks!
